addons/hr_holidays/wizard/hr_departure_wizard.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime, timedelta
+
+from odoo import api, fields, models
+
+
+class HrDepartureWizard(models.TransientModel):
+    _inherit = 'hr.departure.wizard'
+
+    cancel_leaves = fields.Boolean("Cancel Future Leaves", default=True)
+
+    def action_register_departure(self):
+        """To cancel furture leaves, set the state of leaves with the start date later than the 
+        departure date to 'cancel',and for the leaves containts the departure days, change their 
+        end date to the depature day."""
+        super(HrDepartureWizard, self).action_register_departure()
+        if self.cancel_leaves:
+            leaves_to_cancel = self.env['hr.leave'].search([('employee_id', '=', self.employee_id.id), 
+                                                            ('date_from', '>', self.departure_date),
+                                                            ('state', 'not in', ['cancel', 'refuse'])])
+            leaves_to_cancel.sudo().write({'state': 'cancel'})
+            leaves_to_modify = self.env['hr.leave'].search([('employee_id', '=', self.employee_id.id), 
+                                                            ('date_from', '<=', self.departure_date),
+                                                            ('date_to', '>', self.departure_date),
+                                                            ('state', 'not in', ['cancel', 'refuse'])])
+            # In case the start date and end date are the same day, add 17 hours to make sure end date 
+            # is later than the start date
+            end_date  = datetime(self.departure_date.year, 
```

Comment:
I'm not sure about this one. If the leave is overlapping the departure date, I think that it could be better to cancel it too.

The employee will be notified about this, and could take the correct action (Recreate a new one, or not).

Because if the leave is still taken (in Belgium), you loose the money or the time off in your next company.

Is it ok for you ?